pinot-core/src/main/java/org/apache/pinot/core/io/writer/ForwardIndexWriter.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.io.writer;
+
+import java.io.Closeable;
+
+
+/**
+ * Interface for forward index writer.
+ * <p>TODO: Remove this interface and merge the writer implementations into the ForwardIndexCreator implementations.
```

Comment:
This doesn't sound good. Why are we adding it now if we already know it will be deleted in the future?

On that note, why do we need ForwardIndexCreator and ForwardIndexWriter? We might as well use the writer directly. The segment generation code instantiates creators -> creators instantiate writer. Can we just use writer or one of them whichever is suitable?